packages/x-components/src/x-modules/url/components/url-handler.vue

```diff
@@ -99,21 +110,63 @@
     /**
      * Emits the {@link UrlXEvents.ParamsLoadedFromUrl} XEvent,
      * the {@link UrlXEvents.ExtraParamsLoadedFromUrl} XEvent and, if there is query, also emits
-     * the  {@link XEventsTypes.UserOpenXProgrammatically}.
+     * the {@link XEventsTypes.UserOpenXProgrammatically}.
      *
      * @internal
      */
     protected emitEvents(): void {
       const { all, extra } = this.parseUrlParams();
-      this.$x.emit('ParamsLoadedFromUrl', all, { feature: 'url' });
-      this.$x.emit('ExtraParamsLoadedFromUrl', extra, { feature: 'url' });
+      const metadata = { ...this.createOrigin() };
+      this.$x.emit('ParamsLoadedFromUrl', all, metadata);
+      this.$x.emit('ExtraParamsLoadedFromUrl', extra, metadata);
       // TODO: Move this logic from here.
       if (all.query) {
-        this.$x.emit('UserOpenXProgrammatically');
+        this.$x.emit('UserOpenXProgrammatically', undefined, metadata);
       }
       this.urlLoaded = true;
     }
 
+    /**
+     * Creates the origin for the emitted {@link XEvent | XEvents}.
+     *
+     * @returns The metadata.
+     * @internal
+     */
+    protected createOrigin(): QueryOriginInit {
```

Comment:
I would name this just `createWireMetadata`, you are not creating an origin, but an object that can be used to create an origin